src/character.cpp

```diff
@@ -2369,6 +2369,13 @@ void Character::practice( const skill_id &id, int amount, int cap, bool suppress
         return;
     }
 
+    // Your ability to "catch up" skill experience to knowledge is mostly a function of intelligence,
+    // but perception also plays a role, representing both memory/attentiveness and catching on to how
```

Comment:
Would perception really be involved here? Wouldn't something like natural physical prowess (dexterity = muscle memory?) tie to more physical/reflex based skills? Intelligence to skills requiring more complex understanding of concepts and ideas? I'm a bit unclear as to how perception factors in here... attentiveness is more of a factor of focus and mental fortitude (intelligence?) isn't it?

Now if we were talking about learning something through observation of someone demonstrating something to you, then both perception (ability to notice fine details) and intelligence (ability to understand those details) would make sense.

Reply:
Focus, yes. Attentiveness? Plenty of people are intelligent but have problems paying attention - ADHD/ADD. Differentiating between skills in how they're affected by this split is planned.